polaris/polaris/utils.py

```diff
@@ -109,19 +109,32 @@ def verify_valid_asset_operation(
         )
 
 
-def create_stellar_deposit(
-    transaction: Transaction, destination_exists: bool = False
-) -> bool:
+def create_stellar_deposit(transaction: Transaction) -> bool:
     """
     Create and submit the Stellar transaction for the deposit.
-
-    The Transaction can be either `pending_anchor` if the task is called
-    from `poll_pending_deposits()` or `pending_trust` if called from the
-    `check_trustlines()`.
+    The Transaction `pending_anchor` when the task is called from `poll_pending_deposits()`.
+    The Transaction `pending_trust` when the task is called from `check_trustlines()`.
+    The Transaction `pending_anchor_claimable_start` when the task is called from `poll_pending_deposits()`
+        and the Transaction is marked as pending_anchor_claimable_start.
```

Comment:
This is a bit redundant, seems like this case can be removed and the transaction should be `pending_anchor` at this point. Then in this function, we determine if we need to use CBs based on `claimable_balance_supported` and whether or not there is a trust-line, which we also determine in this function.

Reply:
So I updated this logic so it actually is still pending_trust however since `claimable_balance_supported` is True we'll "do the right thing" and craft a claimable balance operation and submit the transaction.
I've detailed the flow in the docstring